### PR TITLE
 Implement a modal for the 'start application' buttons

### LINKED ISSUE #147: Inform users of .gov requirements before starting a new domain request
### Issue description

This ticket is for implementing the designs for https://github.com/cisagov/manage.get.gov/issues/547, also known as the preamble and only for the beta.get.gov portion

Flow from **_beta.get.gov_**:
- When user clicks a button or link to start a new request, a modal appears displaying the Login requirements. 
- Upon logging in (now on manage.get.gov) user is immediately taken to the request flow (rather than the dashboard)

in the related ticket ([#1485](https://github.com/cisagov/manage.get.gov/issues/1485)) on manage.get.gov, a page will need to be added on the application when a user clicks these links/buttons. This needs to be considered during implementation
### Acceptance criteria

- [x] Users clicking the button/links to **start a domain request** from beta.get.gov are shown info about requirements to complete the form, including Login.gov requirements (see the text in additional context)
- [x] after clicking the button/links mentioned and the user moves past the modal + logs in, then they are redirected to a new application 
- [ ] ground work is set for #[1485](https://github.com/cisagov/manage.get.gov/issues/1485) so that when this button/link is clicked a page can be shown conditionally.
- [ ] Don't merge until we are ready to launch

### Additional context

- [New text for both flows](https://docs.google.com/document/d/15FpRLKXGeeNJ5D8uCu1YtRDZSWN5ZrJPKXJV4tCY6fw/edit#heading=h.cldogb82du8p)
- [Mural with flows](https://app.mural.co/t/gsa6/m/gsa6/1701893071504/3b10ecf79684c8e3a1387d7fe5665b58f3c156fa?sender=michellerago0372)
- Figma mockup: [Flow from beta.get.gov](https://www.figma.com/proto/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?page-id=2913%3A27400&type=design&node-id=2927-42517&viewport=131%2C74%2C0.14&t=l70YtUOmnoE59X5H-1&scaling=min-zoom&starting-point-node-id=2927%3A42517&show-proto-sidebar=1&mode=design) and [flow from manage.get.gov](https://www.figma.com/proto/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?page-id=2913%3A27400&type=design&node-id=2926-28192&viewport=131%2C74%2C0.14&t=VIQ4EPkC3bwHn9Xk-1&scaling=min-zoom&starting-point-node-id=2926%3A28192&show-proto-sidebar=1&mode=design)

Note: Some of the text in the document above includes  text for a new page is inserted before the "org type" step of the request flow that will be added in the related ticket. See the whole overall flow in the mural. 

Don't merge until the launch date as this allows a new domain request to be started, we don't want that until we are ready. 

If you need help finding all buttons/links, reach out to a designer for assistance.

### Links to other issues

Design ticket: https://github.com/cisagov/manage.get.gov/issues/547
Related to: cisagov/manage.get.gov#1485

### PR DESCRIPTION
**Pull request by rachidatecs**
## Ticket

Resolves #147 

Linked to PR [#1288](https://github.com/cisagov/manage.get.gov/pull/1621) on manage.get.gov

## Changes

- Added shortcode for a modal for the Start your application links.
- CSS for the a.usa-link
- Inserted shorcode on the following pages:
  - pages/domains/domains_before.md
  - pages/domains/domains_election.md
  - pages/domains/domains_eligibility.md
  - pages/domains/domains_index.md
  - pages/help/help_domain-requests.md

## Context for reviewers

Devs: The whole modal is hardcoded as one shortcode block. This is a small tech debt. If we implement more modals in the future, we'll have to revise to make the shortcode reusable with passed params.

Designers: The typography on the modal header is slightly different than Figma. This is what pulled from the scss codebase (and what I assume is the style of h2s throughout the site)

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [x] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

## Screenshots

Screen Shot 2023-12-30 at 12 42 04 AM

### CODE REVIEW

File: .eleventy.js [hunk 1]
```diff
@@ -159,6 +159,79 @@ module.exports = function (config) {
...
+                    <li>Before you request your first .gov domain, you must verify your identity with Login.gov. This is a necessary layer of security that requires you to prove you are you, and not someone pretending to be you. <a href="https://beta.get.gov/help/account-management/">Get help with Login.gov</a>.</li>
+                  </ul>
+                </li>
+              </ul>
+              <p>If you have <a href="https://beta.get.gov/domains/before/#information-you%E2%80%99ll-need-to-complete-the-domain-request-form">all the information you need</a>, completing your domain request might take around 15 minutes. <a href="https://beta.get.gov/contact/">Contact us if you need help with your request</a>.</p>         
+            </div>
+            <div class="usa-modal__footer">
+              <ul class="usa-button-group">
+                <li class="usa-button-group__item">
+                  <a class="usa-button" href="https://manage.get.gov/register/">
```
Comment (abroddrick) [inline .eleventy.js:204]: for testing this, maybe we should associated it with another sandbox that has @dave-kennedy-ecs code that is the follow on portion of this. That way 1.) when testing we don't accidentally start a real application on manage.get.gov and 2.) we can verify both are working together just fine.

Comment (rachidatecs) [reply .eleventy.js:204]: Looks like it's no longer on the dk sandbox. I think we can get the same test result by typing in the url https://getgov-dk.app.cloud.gov/register/ or even on a local branch http://localhost:8080/register/

**Review by neilmb — COMMENTED** [review]
All this looks reasonable. I think we might want to include a more explicit note in the documentation to tell people to use `modal-start-application` going forward instead of "start a request".

**Comment by gabydisarli** [comment]
@rachidatecs you're right, the modal I've been pulling in and using in Figma is USWDS's modal. Let's go with scss we use on our site, however, can we change the headline to our H3, not the H2? The H3 matches more closely with what USWDS's modal uses.

**Comment by rachidatecs** [comment]
@gabydisarli semantically (and for accessibility/structure) the header on the modal has to be an h2, but I can make that particular h2 look like an h3.

File: .eleventy.js [hunk 2]
```diff
@@ -159,6 +159,79 @@ module.exports = function (config) {
...
+      >
+        <div class="usa-modal__content">
+          <div class="usa-modal__main">
+            <h2 class="usa-modal__heading utility-header" id="modal-start-application-heading">
+            Requirements for requesting a .gov domain
+            </h2>
+            <div class="usa-prose">
+              <ul>
+                <li>You must be a government employee, or be working on behalf of a government organization, to request a .gov domain.</li>
+                <li><a href="https://login.gov/help/get-started/create-your-account/">You must have a Login.gov account</a>. Login.gov provides a simple and secure process for signing into many government services with one account.
```
Comment (therealslimhsiehdy) [inline .eleventy.js:193]: **Question:** I couldn't get the full Figma to load bc of access issues, so bear with me on this (see screenshot below of what I've pulled from Mural):

1. It looks like "You must have a Login.gov account" is bolded, but only "Login.gov account" has a link associated with it
2. The text for the white open circle bullet point is different as well with "Read more about why..." than what you have here

Is Mural the correct wording, or would you mind pasting a screenshot of the Figma with the approved wording? Sorry for running you around in circles!

Screenshot 2024-01-12 at 1 57 11 PM

**Review by therealslimhsiehdy — COMMENTED** [review]
Just have one big question about the wording, otherwise the code and everything looks one and can approve!

Screenshot 2024-01-12 at 1 53 57 PM

File: .eleventy.js (hunk 2 above)
Comment (rachidatecs) [reply .eleventy.js:193]: The approved wording came from the Google doc linked in the ticket. I'm pretty sure I asked @michelle-rago which text is canonical when I was doing this ticket.

**Comment by erinysong** [comment]
nonblocking question - I'm unable to access the Figma wireframes for the modal linked in #147 (currently they lead to an [empty Figma link](https://www.figma.com/proto/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?page-id=2913%3A27400&type=design&node-id=2927-42517&viewport=131%2C74%2C0.14&t=l70YtUOmnoE59X5H-1&scaling=min-zoom&starting-point-node-id=2927%3A42517&show-proto-sidebar=1&mode=design)). I'm only able to find the mocks for manage.get.gov so does anyone have a Figma link for this modal?

**Comment by erinysong** [comment]
Ignore if this was already resolved in previous discussion referencing headers, but should this modal follow USWDS styling or our team's separate styling? Noticed that the USWDS modal header is slightly different
image

File: README.md [hunk 3]
```diff
@@ -29,6 +29,10 @@ For more about the template and it's functionality see [here](https://github.com
 
 `npm run dev`
 
+#### Run with 'clean' to force reset the assets
+
+`npm run dev:clean`
```
[the remark this answers is not in the document]
Comment (erinysong) [reply README.md:34]: ^^^^

**Comment by rachidatecs** [comment]
> Ignore if this was already resolved in previous discussion referencing headers, but should this modal follow USWDS styling or our team's separate styling? Noticed that the USWDS modal header is slightly different image

See previous back and forth with Gaby. TL:DR; she approved the design.

**Review by erinysong — APPROVED** [review]
LGTM - the only questions I had were if we were making our styling consistent to USWDS or our own custom styling but the designers may be more helpful for that area. Otherwise, looks good!

**Review by therealslimhsiehdy — APPROVED** [review]
LGTM! Looks like you've got a merge conflict, but everything looks great + thanks for answering my question (double checked with the Google doc and looks great!)

File: .eleventy.js [hunk 4]
```diff
@@ -159,6 +159,79 @@ module.exports = function (config) {
...
+                  <ul>
+                    <li>Before you request your first .gov domain, you must verify your identity with Login.gov. This is a necessary layer of security that requires you to prove you are you, and not someone pretending to be you. <a href="https://beta.get.gov/help/account-management/">Get help with Login.gov</a>.</li>
+                  </ul>
+                </li>
+              </ul>
```
Comment (abroddrick) [inline .eleventy.js:198]: @rachidatecs I think the beta.get.gov links need to be updated here so it doesn't conflict when merged with what @gabydisarli is doing

Comment (rachidatecs) [reply .eleventy.js:198]: Good call. Will change them to get.gov

Comment (rachidatecs) [reply .eleventy.js:198]: Done

**Comment by gabydisarli** [comment]
@rachidatecs we'll need the modal added to 

- /benefits under _We offer support for your .gov domain heading._ There's a link there.
- /about under _We offer support for your .gov domain_, there's a link there. 
- (the above two places might be taken care of in a content block, not sure)

**Review by gabydisarli — APPROVED** [review]

**Review by erinysong — APPROVED** [review]
LGTM!

**Comment by michelle-rago** [comment]
We want to link to the request form from https://github.com/cisagov/getgov-home/edit/beta/_includes/content-blocks/what_we_offer.md. I see that file is causing a conflict here because we moved to relative links. Not sure how best to resolve without screwing something up. @gabydisarli @rachidatecs

**Comment by rachidatecs** [comment]
> We want to link to the request form from https://github.com/cisagov/getgov-home/edit/beta/_includes/content-blocks/what_we_offer.md. I see that file is causing a conflict here because we moved to relative links. Not sure how best to resolve without screwing something up. @gabydisarli @rachidatecs

Use the modal, no? Basically resolve the conflict by choosing this PR

**Comment by michelle-rago** [comment]
> > We want to link to the request form from https://github.com/cisagov/getgov-home/edit/beta/_includes/content-blocks/what_we_offer.md. I see that file is causing a conflict here because we moved to relative links. Not sure how best to resolve without screwing something up. @gabydisarli @rachidatecs
> 
> Use the modal, no? Basically resolve the conflict by choosing this PR

The modal code has an absolute link to https://beta.get.gov/domains/before, right? @rachidatecs Are you saying it's okay to leave that in there? If so, please resolve the conflict however you see fit!

**Comment by rachidatecs** [comment]
> > > We want to link to the request form from https://github.com/cisagov/getgov-home/edit/beta/_includes/content-blocks/what_we_offer.md. I see that file is causing a conflict here because we moved to relative links. Not sure how best to resolve without screwing something up. @gabydisarli @rachidatecs
> > 
> > 
> > Use the modal, no? Basically resolve the conflict by choosing this PR
> 
> The modal code has an absolute link to https://beta.get.gov/domains/before, right? @rachidatecs Are you saying it's okay to leave that in there? If so, please resolve the conflict however you see fit!

I see. I left the modal in but fixed the absolute link. Just pushed.

**Pull request merged**